harness/features/allFeatures.local.test.ts

```diff
@@ -0,0 +1,92 @@
+import { 
+    getConnectionStringForProxy, 
+    forEachSDK, 
+    describeIf, 
+    createUser,
+    TestClient
+} from '../helpers'
+import { Capabilities, SDKCapabilities } from '../types'
+import { v4 as uuidv4 } from 'uuid'
+import { getServerScope } from '../nock'
+import { config, expectedFeaturesVariationOn } from '../mockData'
+
+jest.setTimeout(10000)
+
+const scope = getServerScope()
+
+describe('allFeatures Tests - Local', () => {
+    forEachSDK((name) => {
+        let url: string
+        const testClient = new TestClient(name)
+        const capabilities: string[] = SDKCapabilities[name]
+        
+        let variationOnUser: string
+        let noVariationUser: string
+        let invalidUser: string
+
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)  
+            
+            await testClient.createClient()
+
+            variationOnUser = (
+                await createUser(url, { user_id: 'user1', customData: { 'should-bucket': true } })
+            ).headers.get('location')
+
+            noVariationUser = (
+                await createUser(url, { user_id: 'user3' })
+            ).headers.get('location')
+
+            invalidUser = (
+                await createUser(url, { name: 'invalid' })
+            ).headers.get('location')
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            describe('uninitialized client', () => {
+                it('should return empty object if client is uninitialized',  async () => {
+                    const featuresResponse = await testClient.callAllFeatures(variationOnUser)
```

Comment:
any tests like this should probably use their own client that is guaranteed to not have initialized by now. The problem with sharing one client for this whole suite of tests is that Jest doesn't guarantee what order the individual test cases run in. You could get unlucky and have the tests that wait for the client to be initialized run first.

Reply:
oh true good point, i forgot to fix that in this pr - fixed!